src/plugin.cc

```diff
@@ -409,14 +482,48 @@ bool plugin::Zeek_Spicy::Plugin::togglePacketAnalyzer(const ::spicy::zeek::compa
     if ( type >= _packet_analyzers_by_type.size() )
         return false;
 
-    const auto& analyzer = _protocol_analyzers_by_type[type];
+    const auto& analyzer = _packet_analyzers_by_type[type];
 
     if ( ! analyzer.type )
         // not set -> not ours
         return false;
-    ZEEK_DEBUG(hilti::rt::fmt("supposed to toggle packet analyzer %s, but that is not supported by Zeek",
+
+#if ZEEK_VERSION_NUMBER >= 50200
+    ::zeek::packet_analysis::Component* component = ::zeek::packet_mgr->Lookup(tag);
+    ::zeek::packet_analysis::Component* component_replaces =
+        analyzer.replaces ? ::zeek::packet_mgr->Lookup(analyzer.replaces) : nullptr;
+
+    if ( ! component ) {
+        // Shouldn't really happen.
+        reporter::internalError("failed to lookup packet analyzer component");
+        return false;
+    }
+
+    if ( enable ) {
+        ZEEK_DEBUG(hilti::rt::fmt("Enabling Spicy packet analyzer %s", analyzer.name_analyzer));
+        component->SetEnabled(true);
+
+        if ( component_replaces ) {
+            ZEEK_DEBUG(hilti::rt::fmt("Disabling standard packet analyzer %s", analyzer.name_analyzer));
```

Comment:
Same pattern as elsewhere, but I think we don't really know whether we build a Zeek analyzer (it could be any previously loaded one).